File: app/src/main/java/com/tohami/egyptinnovate/app/localization/LanguageSelectionInspector.java
package com.tohami.egyptinnovate.app.localization;

import com.tohami.egyptinnovate.BuildConfig;
import com.tohami.egyptinnovate.app.settings.AppSettings;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;

public class LanguageSelectionInspector implements Interceptor {
    private volatile AppSettings appSettings;

    public void setAppSettings(AppSettings appSettings) {
        this.appSettings = appSettings;
    }

    @NotNull
    @Override
    public okhttp3.Response intercept(@NotNull Chain chain) throws IOException {
        Request request = chain.request();

        String oldUrl = request.url().toString();

        String newUrl ;
        if(appSettings.getCurrentLanguage() == Language.ARABIC) {
            newUrl = oldUrl.replace(BuildConfig.BASE_EN_URL, BuildConfig.BASE_AR_URL);
        }else {
            newUrl = oldUrl.replace(BuildConfig.BASE_AR_URL, BuildConfig.BASE_EN_URL);
        }

        request = request.newBuilder()
                .url(newUrl)
                .build();

        return chain.proceed(request);
    }
}

File: app/src/main/java/com/tohami/egyptinnovate/app/localization/LanguageSelectionInspector.kt
package com.tohami.egyptinnovate.app.localization

import com.tohami.egyptinnovate.BuildConfig
import com.tohami.egyptinnovate.app.settings.AppSettings
import okhttp3.Interceptor
import java.io.IOException

class LanguageSelectionInspector : Interceptor {
    @Volatile
    private lateinit var appSettings: AppSettings

    fun setAppSettings(appSettings: AppSettings) {
        this.appSettings = appSettings
    }

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): okhttp3.Response {
        chain.request().run {
            val oldUrl = url.toString()

            val newUrl = when(appSettings.currentLanguage) {
                Language.ARABIC -> oldUrl.replace(BuildConfig.BASE_EN_URL, BuildConfig.BASE_AR_URL)
                else -> oldUrl.replace(BuildConfig.BASE_AR_URL, BuildConfig.BASE_EN_URL)
            }

            val newRequest =  newBuilder().url(newUrl).build()
            return chain.proceed(newRequest)
        }
    }
}
